server/src/test/java/org/elasticsearch/gateway/MetadataStateFormatTests.java

```diff
@@ -280,7 +280,6 @@ public void testFailWriteAndReadPreviousState() throws IOException {
 
         for (int i = 0; i < randomIntBetween(1, 5); i++) {
             format.failOnMethods(
-                Format.FAIL_DELETE_TMP_FILE,
```

Comment:
Would be nice to still check that if we need to delete a file and the delete fails then we do the right thing.

Reply:
Sure thing added some infrastructure for this and a test that simulates the delete failure and the catch above in one (with the auto-cleanup on this path I needed both a write and delete failure in one anyway :)).